src/main/java/org/broadinstitute/dsm/route/DownloadPDFRoute.java

```diff
@@ -197,235 +169,226 @@ public static Object returnPDFS(Map<String, Map<String, Object>> participantESDa
-                        }
-                        else if (StringUtils.isNotBlank(startDate) && startDate.startsWith("0/") && !startDate.equals("0/0")) {
-                            startDate = "01/" + startDate.split("/")[1];
-                        }
-                    }
-                    if (keySet.contains(JSON_END_DATE)) {
-                        endDate = (String) jsonObject.get(JSON_END_DATE);
-                        endDate = SystemUtil.changeDateFormat(SystemUtil.DATE_FORMAT, SystemUtil.US_DATE_FORMAT, endDate);
-                    }
-                    if (StringUtils.isBlank(medicalRecordId)) {
-                        throw new RuntimeException("MedicalRecordID is missing. Can't create cover pdf");
-                    }
-                    //get information from db
-                    MedicalRecord medicalRecord = MedicalRecord.getMedicalRecord(realm, ddpParticipantId, medicalRecordId);
-
-                    PDFProcessor processor = new CoverPDFProcessor(ddpInstance.getName());
-                    Map<String, Object> valueMap = new HashMap<>();
-                    //values same no matter from where participant/institution data comes from
-                    valueMap.put(CoverPDFProcessor.FIELD_CONFIRMED_INSTITUTION_NAME, medicalRecord.getName());
-                    valueMap.put(CoverPDFProcessor.FIELD_CONFIRMED_INSTITUTION_NAME_2, medicalRecord.getName() + ",");
-                    valueMap.put(CoverPDFProcessor.FIELD_CONFIRMED_FAX, medicalRecord.getFax());
-                    String today = new SimpleDateFormat("MM/dd/yyyy").format(new Date());
-                    valueMap.put(CoverPDFProcessor.FIELD_DATE, today);
-                    valueMap.put(CoverPDFProcessor.FIELD_DATE_2, StringUtils.isNotBlank(endDate) ? endDate : today); //end date
-
-                    //adding checkboxes configured under instance_settings
-                    InstanceSettings instanceSettings = InstanceSettings.getInstanceSettings(realm);
-                    if (instanceSettings != null && instanceSettings.getMrCoverPdf() != null && !instanceSettings.getMrCoverPdf().isEmpty()) {
-                        for (Value mrCoverSetting : instanceSettings.getMrCoverPdf()) {
-                            if (keySet.contains(mrCoverSetting.getValue())) {
-                                valueMap.put(mrCoverSetting.getValue(), BooleanUtils.toBoolean((Boolean) jsonObject.get(mrCoverSetting.getValue())));
+    public byte[] getPDFBundle(@NonNull String ddpParticipantId, String medicalRecordId,
+                             List<String> oncHistoryIDs, List<PDF> pdfs, @NonNull String realm, String requestBody) {
+        DDPInstance ddpInstance = DDPInstance.getDDPInstance(realm);
+        if (ddpInstance != null && StringUtils.isNotBlank(ddpParticipantId)) {
+            ByteArrayOutputStream output = new ByteArrayOutputStream();
```

Comment:
try to always close resource streams either using try-with-resource or their `close()` method, try-with-resource is always preferable.